Automattic/wp-calypso
# Stats: Extracting redirect into a new component

Related to #86132

## Proposed Changes

* Refactor solution from #86211 and place redirecting code in a separate component
* Enhance purchase flow condition with a flag from an endpoint

## Testing Instructions



* Apply D134498 to your sandbox
* Navigate to a blog created last year and verify that the stats page loads
* Apply `stats/checkout-flows-v2` and verify that the page still loads the same way
* Navigate to a new JN page with the feature flag applied and verify that it redirects to the purchase flow
* Go to the Stats page for that JN blog, but without the feature flag applied and verify that the stats page loads (not the purchase page)

## Pre-merge Checklist
